testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -505,6 +508,32 @@ def test_fasta2select_ClustalW(self, tmpdir):
         assert len(
             sel['mobile']) == 23090, "selection string has unexpected length"
 
+    def test_fasta2select_nogap(self, tmpdir):
+        """test align.fasta2select() on aligned FASTA with no gap in residue
+        (Issue #3124)"""
+        reference = mda.Universe(self.ref).select_atoms('all')
+        target = mda.Universe(self.tgt).select_atoms('all')
+        ref_resids = [a.resid for a in
+                      reference.select_atoms('protein').select_atoms(
+                          'name CA and not altloc B')]
+        target_resids = [a.resid for a in
+                         target.select_atoms('protein').select_atoms(
+                             'name CA and not altloc B')]
+        alignment = align.sequence_alignment(
+            target.select_atoms('protein and not altloc B'),
+            reference.select_atoms('protein and not altloc B'))
+        with tmpdir.as_cwd():
+            fasta = f">ref\n{alignment[0]}\n>mobile\n{alignment[1]}"
+            with open('alignment.aln', "w+") as f:
+                f.write(fasta)
+            sel = align.fasta2select('alignment.aln',
+                                     is_aligned=True,
+                                     ref_resids=ref_resids,
+                                     target_resids=target_resids)
+        assert len(sel['reference']) == 5455, ("selection string has"
```

Comment:
given the number of times we write the words "selection string has unexpected lenth" (often using up multiple lines), maybe it's worth just declaring as the one class variable? (fewer lines in codebase is always better)